File: src/backend/mdfe/src/model/mdfe_informacao_cte_model.py
from src import db
from src.model.mdfe_municipio_descarrega_model import MdfeMunicipioDescarregaModel


class MdfeInformacaoCteModel(db.Model):
    __tablename__ = 'mdfe_informacao_cte'

    id = db.Column(db.Integer, primary_key=True)
    chave_cte = db.Column(db.String(44))
    segundo_codigo_barra = db.Column(db.String(36))
    indicador_reentrega = db.Column(db.Integer)
    id_mdfe_municipio_descarrega = db.Column(db.Integer, db.ForeignKey('mdfe_municipio_descarrega.id'))

    mdfe_municipio_descarrega_model = db.relationship('MdfeMunicipioDescarregaModel', foreign_keys=[id_mdfe_municipio_descarrega])

    def mapping(self, data):
        self.id = data.get('id')
        self.id_mdfe_municipio_descarrega = data.get('idMdfeMunicipioDescarrega')
        self.chave_cte = data.get('chaveCte')
        self.segundo_codigo_barra = data.get('segundoCodigoBarra')
        self.indicador_reentrega = data.get('indicadorReentrega')

    def serialize(self):
        return {
            'id': self.id,
            'idMdfeMunicipioDescarrega': self.id_mdfe_municipio_descarrega,
            'chaveCte': self.chave_cte,
            'segundoCodigoBarra': self.segundo_codigo_barra,
            'indicadorReentrega': self.indicador_reentrega,
            'mdfeMunicipioDescarregaModel': self.mdfe_municipio_descarrega_model.serialize() if self.mdfe_municipio_descarrega_model else None,
        }

File: frontend/mdfe/lib/app/data/model/mdfe_informacao_cte_model.dart
import 'dart:convert';
import 'package:pluto_grid/pluto_grid.dart';

import 'package:mdfe/app/data/model/model_imports.dart';

class MdfeInformacaoCteModel {
	int? id;
	int? idMdfeMunicipioDescarrega;
	String? chaveCte;
	String? segundoCodigoBarra;
	int? indicadorReentrega;
	MdfeMunicipioDescarregaModel? mdfeMunicipioDescarregaModel;

	MdfeInformacaoCteModel({
		this.id,
		this.idMdfeMunicipioDescarrega,
		this.chaveCte,
		this.segundoCodigoBarra,
		this.indicadorReentrega,
		this.mdfeMunicipioDescarregaModel,
	});

	static List<String> dbColumns = <String>[
		'id',
		'chave_cte',
		'segundo_codigo_barra',
		'indicador_reentrega',
	];
	
	static List<String> aliasColumns = <String>[
		'Id',
		'Chave Cte',
		'Segundo Codigo Barra',
		'Indicador Reentrega',
	];

	MdfeInformacaoCteModel.fromJson(Map<String, dynamic> jsonData) {
		id = jsonData['id'];
		idMdfeMunicipioDescarrega = jsonData['idMdfeMunicipioDescarrega'];
		chaveCte = jsonData['chaveCte'];
		segundoCodigoBarra = jsonData['segundoCodigoBarra'];
		indicadorReentrega = jsonData['indicadorReentrega'];
		mdfeMunicipioDescarregaModel = jsonData['mdfeMunicipioDescarregaModel'] == null ? MdfeMunicipioDescarregaModel() : MdfeMunicipioDescarregaModel.fromJson(jsonData['mdfeMunicipioDescarregaModel']);
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonData = <String, dynamic>{};

		jsonData['id'] = id != 0 ? id : null;
		jsonData['idMdfeMunicipioDescarrega'] = idMdfeMunicipioDescarrega != 0 ? idMdfeMunicipioDescarrega : null;
		jsonData['chaveCte'] = chaveCte;
		jsonData['segundoCodigoBarra'] = segundoCodigoBarra;
		jsonData['indicadorReentrega'] = indicadorReentrega;
		jsonData['mdfeMunicipioDescarregaModel'] = mdfeMunicipioDescarregaModel?.toJson;
	
		return jsonData;
	}
	
	String objectEncodeJson() {
		final jsonData = toJson;
		return json.encode(jsonData);
	}

	plutoRowToObject(PlutoRow plutoRow) {
		id = plutoRow.cells['id']?.value;
		idMdfeMunicipioDescarrega = plutoRow.cells['idMdfeMunicipioDescarrega']?.value;
		chaveCte = plutoRow.cells['chaveCte']?.value;
		segundoCodigoBarra = plutoRow.cells['segundoCodigoBarra']?.value;
		indicadorReentrega = plutoRow.cells['indicadorReentrega']?.value;
		mdfeMunicipioDescarregaModel = MdfeMunicipioDescarregaModel();
		mdfeMunicipioDescarregaModel?.nomeMunicipio = plutoRow.cells['mdfeMunicipioDescarregaModel']?.value;
	}	

	MdfeInformacaoCteModel clone() {
		return MdfeInformacaoCteModel(
			id: id,
			idMdfeMunicipioDescarrega: idMdfeMunicipioDescarrega,
			chaveCte: chaveCte,
			segundoCodigoBarra: segundoCodigoBarra,
			indicadorReentrega: indicadorReentrega,
		);			
	}

	
}